### DISCUSSION TITLE
New release v1.3.0 🚀

### DISCUSSION THREAD

**Discussion by diegoferigo** [robotology/gym-ignition#401] [Releases]
Compatible with [Ignition **Fortress**](https://ignitionrobotics.org/docs/fortress/install) (`Gazebo ≥ 6.0.0`).

## Bug fixing and new features

- Bump Ignition distribution to Edifice (#307)
- Align Physics system (#318)
- Address sdformat11 deprecations (#321)
- Align empty world with upstream (#336)
- Simplify passing the physics parameters to the `Server` (#335)
- Improve how links with no collision elements are handled in contact detection (#334)
- Prepare multiple physics engines support (#338)
- Move inheritance from `ModelWithFile` to `ICubGazeboABC` (#342)
- Automatic configuration of a sane default verbosity (#343)
- Allow loading link and joint plugins (#345)
- Fix call to `setComTarget` in InverseKinematicsNLP (#339)
- Split `scenario` and `gym-ignition` Python packages (#346)
- Support setting joint velocity limits (#356)
- Fix type hinting of binded methods returning `std::vector`s (#357)
- Fix include directories in SWIG bindings (#358)
- Update vendored FindPython resources to CMake 3.20.2 (#360)
- Update setuptools resources introducing `setup.cfg` (#364)
- Align Physics system (Edifice 5.1) (#365)
- Update vendored FindPython3 to CMake 3.20.5 (#367)
- Use the versioned name of the DART physics plugin (#369)
- Update the PyPI name to `scenario` (#373)
- Align Physics system (#376)
- Format Python files with black and isort (#377)
- Remove the need to use a singleton to get the ECM pointer (#379)
- Fix leftover after removing `ScenarioGazeboPlugins` CMake component (#382)
- Disable joint velocity limits in the manipulation example (#384)
- Allow inserting worlds from file containing models (#386)
- Fix `test_velocity_direct` (#387)
- Allow specifying different pos and rot weights for POSE IK targets  (#388)
- Application of external wrenches: new `empty` method and new Python test (#391)
- Minor cleanup (#395)
- Align Physics system (Fortress) (#392)
- Check for a compatible Ignition distribution in the system when importing the `scenario` Python package (#399)
- Fix positivity option of model randomizations (#400)
- Bump Ignition distribution to Fortress (#383)
- [`devel`]→[`master`] Preparation of v1.3.0 (#361) 

## Documentation, CI/CD, tests

- Add issue templates (#352)
- Test a multi-world simulation with DART and bullet collision detector (#368)
- Disable `test_multi_world_simulation` (#374)
- Enforce compliance of clang-format style (#375)
- Create CODEOWNERS (#380)

### Python Packages

This release includes #346 that changes how we distribute the project in PyPI. Starting with `v1.3.0`, the PyPI package `gym-ignition` will only provide the Python resources of the `gym_ignition` module (and the demo `gym_ignition_environments`). **A new `scenario` PyPI package will provide the C++ resources.** The import names didn't change (`import gym_ignition`, `import scenario`), and downstream should not be affected in most cases. 

**Warning:** if a downstream Python package was using the exported CMake targets of ScenarIO, some maintenance action is necessary. These projects have `gym_ignition` in the `[build-system]/requires` of their `pyproject.toml`. 

In order to update to this `v1.3.0` release, maintainers have to take the following action:

- **`pyproject.toml`**: replace `gym_ignition` with `scenario ~= 1.3.0`.

### API change

This time, in https://github.com/robotology/gym-ignition/pull/321, we performed a minor update of our public APIs. If you used the following function:

```cpp
bool  scenario::gazebo::utils::renameSDFModel(sdf::Root& sdfRoot, 
                                              const std::string& newModelName, 
                                              const size_t modelIndex = 0);
```

please note that the `modelIndex` defaulted argument has been removed. The new signature is the following:

```cpp
bool  scenario::gazebo::utils::renameSDFModel(sdf::Root& sdfRoot,
                                              const std::string& newModelName);
```

## Contributors

- @diegoferigo 
- @traversaro

<hr /><em>This discussion was created from the release <a href='https://github.com/robotology/gym-ignition/releases/tag/v1.3.0'>v1.3.0</a>.</em>

**Comment by AndrejOrsula** [comment]
Thank you once again for the great work! I can confirm that local (development) build of [v1.3.0.post0](https://github.com/robotology/gym-ignition/releases/tag/v1.3.0.post0) works out of the box for me and I have not noticed any issues yet.

**Reply by diegoferigo** [reply]
Thanks Andrej for the feedback, much appreciated! It's important from our side knowing that things didn't break, the jump Dome → Fortress was pretty big.
